Framework/DataObjects/src/RebinnedOutput.cpp

```diff
@@ -132,6 +137,35 @@ void RebinnedOutput::finalize(bool hasSqrdErrs) {
               std::ostream_iterator<double>(g_log.debug(), " "));
     g_log.debug() << '\n';
   }
+  // Sets flag so subsequent algorithms know to correctly treat data
+  m_finalized = true;
+}
+
+/**
+ * This function "unfinalizes" the workspace by taking the data/error arrays
+ * and multiplying them by the corresponding fractional area array.
+ * @param hasSqrdErrs :: does the workspace have squared errors?
+ */
+void RebinnedOutput::unfinalize(bool hasSqrdErrs) {
+  if (!m_finalized) {
+    return;
+  }
+  std::size_t nHist = this->getNumberHistograms();
+  for (std::size_t i = 0; i < nHist; ++i) {
```

Comment:
Can we do this in parallel?

Reply:
I don't know - is `std::transform` thread safe. Someone on SO implies that it is not, but I can't find anything more concrete...

Reply:
As long as it is treating the histograms independently it will be safe.